flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/com/ververica/cdc/connectors/mysql/factory/MySqlDataFactory.java

```diff
@@ -0,0 +1,367 @@
+
+        boolean scanNewlyAddedTableEnabled =
+                config.get(MySqlDataSourceOptions.SCAN_NEWLY_ADDED_TABLE_ENABLED);
+        boolean includeSchemaChangeEnabled =
+                config.get(MySqlDataSourceOptions.INCLUDE_SCHEMA_CHANGE_ENABLED);
+
+        int fetchSize = config.get(MySqlDataSourceOptions.SCAN_SNAPSHOT_FETCH_SIZE);
+        int splitSize = config.get(MySqlDataSourceOptions.SCAN_INCREMENTAL_SNAPSHOT_CHUNK_SIZE);
+        int splitMetaGroupSize = config.get(MySqlDataSourceOptions.CHUNK_META_GROUP_SIZE);
+        String chunkKeyColumn =
+                config.getOptional(
+                                MySqlDataSourceOptions.SCAN_INCREMENTAL_SNAPSHOT_CHUNK_KEY_COLUMN)
+                        .orElse(null);
+        double distributionFactorUpper =
+                config.get(MySqlDataSourceOptions.CHUNK_KEY_EVEN_DISTRIBUTION_FACTOR_UPPER_BOUND);
+        double distributionFactorLower =
+                config.get(MySqlDataSourceOptions.CHUNK_KEY_EVEN_DISTRIBUTION_FACTOR_LOWER_BOUND);
+
+        boolean closeIdleReaders =
+                config.get(MySqlDataSourceOptions.SCAN_INCREMENTAL_CLOSE_IDLE_READER_ENABLED);
+
+        Duration heartbeatInterval = config.get(MySqlDataSourceOptions.HEARTBEAT_INTERVAL);
+        Duration connectTimeout = config.get(MySqlDataSourceOptions.CONNECT_TIMEOUT);
+        int connectMaxRetries = config.get(MySqlDataSourceOptions.CONNECT_MAX_RETRIES);
+        int connectionPoolSize = config.get(MySqlDataSourceOptions.CONNECTION_POOL_SIZE);
+
+        validateIntegerOption(
+                MySqlDataSourceOptions.SCAN_INCREMENTAL_SNAPSHOT_CHUNK_SIZE, splitSize, 1);
+        validateIntegerOption(MySqlDataSourceOptions.CHUNK_META_GROUP_SIZE, splitMetaGroupSize, 1);
+        validateIntegerOption(MySqlDataSourceOptions.SCAN_SNAPSHOT_FETCH_SIZE, fetchSize, 1);
+        validateIntegerOption(MySqlDataSourceOptions.CONNECTION_POOL_SIZE, connectionPoolSize, 1);
+        validateIntegerOption(MySqlDataSourceOptions.CONNECT_MAX_RETRIES, connectMaxRetries, 0);
+        validateDistributionFactorUpper(distributionFactorUpper);
+        validateDistributionFactorLower(distributionFactorLower);
+
+        Map<String, String> configMap = config.toMap();
+        OptionUtils.printOptions(IDENTIFIER, config.toMap());
+
+        MySqlSourceConfigFactory configFactory =
+                new MySqlSourceConfigFactory()
+                        .hostname(hostname)
+                        .port(port)
+                        .username(username)
+                        .password(password)
+                        .databaseList(".*")
+                        .tableList(".*")
+                        .startupOptions(startupOptions)
+                        .serverId(serverId)
+                        .serverTimeZone(serverTimeZone.getId())
+                        .fetchSize(fetchSize)
+                        .splitSize(splitSize)
+                        // TODO: .chunkKeyColumn()
```

Comment:
we can create an issue to support chunKeyColumn in table level in mysql cdc connector